feat: add orderBy functionality to BaseRepository

dagmawig opened the pull request:
fixes issue #37 

![image](https://user-images.githubusercontent.com/61099792/150279249-f3ead5e2-f006-4629-8b52-14320ed3ee8c.png)

dadiorchen reviewed (changes requested):
@dagmawig I need you to do some change:
1. Replace line 76 with a complete SQL sentence, which means we don't use '*' to jump text anymore.
2. Create an another test with name: 'getByFilter without order', in this test you shouldn't set the order, and also, check the result with complete SQL.

dadiorchen commented on server/infra/database/BaseRepository.spec.ts:
```diff
@@ -69,6 +69,25 @@ describe('BaseRepository', () => {
+      tracker.uninstall()
+      tracker.install()
+      tracker.on('query', (query) => {
+        expect(query.sql).toMatch(/select.*testTable.*name.*order.*by.*desc.*/)
```
Before we didn't use the complete SQL to verify the result is because that the logic is simple, now the logic become more and more complicated, so we need to check it with the complete SQL

dagmawig commented:
I have removed wildcards from the regex expression. I have also added a test for when the orderBy parameter is not provided. 

![image](https://user-images.githubusercontent.com/61099792/150897669-3a7287f3-9a65-4085-8106-34fb2d13ab1b.png)
![image](https://user-images.githubusercontent.com/61099792/150897757-c2d5f5e3-9432-4ff9-937f-6b5086974e95.png)

dadiorchen commented:
Please solve the conflict above @dagmawig it's not your fault, our new merges led to the conflict.

dadiorchen commented on server/infra/database/BaseRepository.spec.ts:
```diff
@@ -69,6 +69,43 @@ describe('BaseRepository', () => {
+      tracker.uninstall()
+      tracker.install()
+      tracker.on('query', (query) => {
+        expect(query.sql).toMatch('select \* from "testTable" where \("name" \= $1\) order by "id" desc'
```
Nice, I thought you don't need to use the backslash to escape the star, unless you are typing regular express.

dadiorchen commented:
@dagmawig thank you!
